pkg/pool-stable-phantom/test/StablePoolAmplification.test.ts

```diff
@@ -0,0 +1,300 @@
+              'MAX_AMP'
+            );
+          });
+
+          describe('rate limits', () => {
+            let startTime: BigNumber;
+
+            beforeEach('set start time', async () => {
+              startTime = (await currentTimestamp()).add(100);
+              await setNextBlockTimestamp(startTime);
+            });
+
+            it('reverts when increasing the amp by more than 2x in a single day', async () => {
+              const newAmp = AMPLIFICATION_PARAMETER.mul(2).add(1);
+              const endTime = startTime.add(DAY);
+
+              await expect(pool.connect(owner).startAmplificationParameterUpdate(newAmp, endTime)).to.be.revertedWith(
+                'AMP_RATE_TOO_HIGH'
+              );
+            });
+
+            it('reverts when increasing the amp by more than 2x daily over multiple days', async () => {
+              const newAmp = AMPLIFICATION_PARAMETER.mul(5).add(1);
+              const endTime = startTime.add(DAY * 2);
+
+              await expect(pool.connect(owner).startAmplificationParameterUpdate(newAmp, endTime)).to.be.revertedWith(
+                'AMP_RATE_TOO_HIGH'
+              );
+            });
+
+            it('reverts when decreasing the amp by more than 2x in a single day', async () => {
+              const newAmp = AMPLIFICATION_PARAMETER.div(2).sub(1);
+              const endTime = startTime.add(DAY);
+
+              await expect(pool.connect(owner).startAmplificationParameterUpdate(newAmp, endTime)).to.be.revertedWith(
+                'AMP_RATE_TOO_HIGH'
+              );
+            });
+
+            it('reverts when decreasing the amp by more than 2x daily over multiple days', async () => {
+              const newAmp = AMPLIFICATION_PARAMETER.div(5).sub(1);
+              const endTime = startTime.add(DAY * 2);
+
+              await expect(pool.connect(owner).startAmplificationParameterUpdate(newAmp, endTime)).to.be.revertedWith(
+                'AMP_RATE_TOO_HIGH'
+              );
+            });
+          });
+        });
+      });
+
+      context('when requesting a short duration change', () => {
+        let endTime;
+
+        it('reverts', async () => {
+          endTime = (await currentTimestamp()).add(DAY).sub(1);
+          await expect(
+            pool.connect(owner).startAmplificationParameterUpdate(AMPLIFICATION_PARAMETER, endTime)
+          ).to.be.revertedWith('AMP_END_TIME_TOO_CLOSE');
+        });
+      });
+    });
+
+    context('when the sender is not allowed', () => {
+      it('reverts', async () => {
+        await expect(pool.connect(other).stopAmplificationParameterUpdate()).to.be.revertedWith('SENDER_NOT_ALLOWED');
+        await expect(
+          pool.connect(other).startAmplificationParameterUpdate(AMPLIFICATION_PARAMETER, DAY)
+        ).to.be.revertedWith('SENDER_NOT_ALLOWED');
+      });
+    });
+  });
+
+  describe('stopAmplificationParameterUpdate', () => {
```

Comment:
We're lacking permission tests for this

Reply:
[d9a297a](https://github.com/balancer-labs/balancer-v2-monorepo/pull/1559/commits/d9a297ac6257f8a7fc34a019b8e9e6c602b5f6b1)